src/tokens/tokens.utils.ts

```diff
@@ -39,15 +39,15 @@ export async function getVaultTokens(
   balance: number,
   currency?: Currency,
 ): Promise<TokenValue[]> {
-  const { protocol, vaultToken, depositToken, getTokenBalance } = vaultDefinition;
-  const token = await thisModule.getFullToken(chain, vaultToken);
+  const { depositToken, getTokenBalance } = vaultDefinition;
+  const token = await thisModule.getFullToken(chain, depositToken);
```

Comment:
this was the bug, using the vault token always meant that it was not defined as an LP or a UniV2 pricing type so only custom token lookups got cached